### PR TITLE
AUQA-93_uiPopupVerificationCX1

### PR DESCRIPTION
**Pull request by abhijitvigs**

### CODE REVIEW

File: PageObjects/siteEditorHomePage.py [hunk 1]
```diff
@@ -15,3 +15,8 @@
 close_button="xpath=//span[contains(text(),'Close')]"
 all_properties_button="xpath=//span[contains(text(),'All Properties')]"
 group_dropdown_list="xpath=//div[1]/div/div[1]/div[2]/div/div[2]/div/div/div[2]"
+tools_configs_system = 'xpath=//div[4]/div/div/div/div/div[2]'
+sfc_min = 'xpath=//table[25]/tbody/tr/td[2]/div'
+tools_configs_dashm = 'xpath=/html/body/div[1]/div[3]/div[2]/div/div/div/div[2]/div/div/div[2]/div[1]/div/div/div[1]/div[2]'
```
Comment (ananiac) [inline PageObjects/siteEditorHomePage.py:20]: remove /html/body and use //

File: PageObjects/siteEditorHomePage.py [hunk 2]
```diff
@@ -15,3 +15,8 @@
...
+turn_off_limit_24_hr = 'xpath=/html/body/div[1]/div[3]/div[2]/div/div/div/div[2]/div/div/div[2]/div[2]/div[2]/div/div[2]/table[23]/tbody/tr/td[2]/div'
```
Comment (ananiac) [inline PageObjects/siteEditorHomePage.py:21]: remove /html/body and use //

**Review by ananiac — COMMENTED** [review]
Update the excel sheet - testexecutioncommands - with pabot command for MinonGeneral-Test and UiTestpopup

Execute the test across all the VEMS

File: PageObjects/siteEditorHomePage.py [hunk 3]
```diff
@@ -15,3 +15,8 @@
 close_button="xpath=//span[contains(text(),'Close')]"
 all_properties_button="xpath=//span[contains(text(),'All Properties')]"
 group_dropdown_list="xpath=//div[1]/div/div[1]/div[2]/div/div[2]/div/div/div[2]"
+tools_configs_system= "xpath=//div[contains(text(),'SYSTEM')]"
+tools_configs_dashm= "xpath=//div[contains(text(),'DASHM') and @unselectable='on']"
+yes_button = "xpath=//span[contains(text(),'Yes')]"
+sfc_min= "xpath=//div[contains(text(),'SFCMin')]//following::td"
```
Comment (ananiac) [inline PageObjects/siteEditorHomePage.py:21]: THis gives 20 element so  select the first specifically

File: PageObjects/siteEditorHomePage.py [hunk 4]
```diff
@@ -15,3 +15,8 @@
...
+turn_off_limit_24_hr= "xpath=//div[contains(text(),'TurnOffLimit24H')]//following::td"
```
Comment (ananiac) [inline PageObjects/siteEditorHomePage.py:22]: gives 125 element - //div[contains(text(),'TurnOffLimit24H')]//following::td. select the desired index

File: Resources/UIResources/uiPopupResources.robot [hunk 5]
```diff
@@ -188,4 +189,119 @@ verifySupressAlramPopupMessageOfFirstRecord
...
+    sleep    ${load_time}
+    ${expected_title}=  set variable  ${test_input}[validation_error_popup_title]
+    verifyPopupTitle  ${popup_title}  ${expected_title}
+    ${expected_message}=  set variable  ${test_input}[turn_off_limit_24_hr_popup_message]
+    verifyPopupMessage  ${popup_message}  ${expected_message}
+    okButtonClick
+    log to console  Title & message of 'Validation Error' (for 'TurnOffLimit24Hr' field) popup verified successfully
+    close browser
+
+unsavedChanges
```
Comment (ananiac) [inline Resources/UIResources/uiPopupResources.robot:244]: Change to verifyPopupUnsavedChanges

File: Resources/UIResources/uiPopupResources.robot [hunk 6]
```diff
@@ -188,4 +189,119 @@ verifySupressAlramPopupMessageOfFirstRecord
...
+    verifyPopupTitle  ${popup_title}  ${expected_title}
+    ${expected_message}=  set variable  ${test_input}[sfcmin_popup_message]
+    verifyPopupMessage  ${popup_message}  ${expected_message}
+    click element  ${ok_button}
+    log to console  'OK' button clicked of 'Validation Error' popup
+    sleep  ${load_time}
+    log to console  Title & message of 'Validation Error' popup verified successfully
+    close browser
+
+savePopup
```
Comment (ananiac) [inline Resources/UIResources/uiPopupResources.robot:212]: Change to verifySavePopup

File: Resources/UIResources/uiPopupResources.robot [hunk 7]
```diff
@@ -188,4 +189,119 @@ verifySupressAlramPopupMessageOfFirstRecord
...
+    saveButtonClick
+    ${expected_title}=  set variable  ${test_input}[save_popup_title]
+    verifyPopupTitle  ${popup_title}  ${expected_title}
+    ${expected_message}=  set variable  ${test_input}[save_popup_message]
+    verifyPopupMessage  ${popup_message}  ${expected_message}
+    okButtonClick
+    log to console  Title & message of 'Save' popup verified successfully
+    close browser
+
+validationErrorForTurnOffLimit24Hr
```
Comment (ananiac) [inline Resources/UIResources/uiPopupResources.robot:227]: verifyValidationErrorForTurnOffLimit24HrsettoNegativeNo

Comment (abhijitvigs) [reply Resources/UIResources/uiPopupResources.robot:227]: done

File: Resources/UIResources/uiPopupResources.robot (hunk 6 above)
Comment (abhijitvigs) [reply Resources/UIResources/uiPopupResources.robot:212]: done

File: Resources/UIResources/uiPopupResources.robot (hunk 5 above)
Comment (abhijitvigs) [reply Resources/UIResources/uiPopupResources.robot:244]: done

File: PageObjects/siteEditorHomePage.py (hunk 4 above)
Comment (abhijitvigs) [reply PageObjects/siteEditorHomePage.py:22]: changed as per discussion

File: PageObjects/siteEditorHomePage.py (hunk 3 above)
Comment (abhijitvigs) [reply PageObjects/siteEditorHomePage.py:21]: changed as per discussion

**Pull request merged**